parts/k8s/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -225,6 +225,22 @@ apt_get_dist_upgrade() {
   echo Executed apt-get dist-upgrade $i times
   wait_for_apt_locks
 }
+apt_fix_keys() {
+  retries=10
+  output=/tmp/apt-fix-keys.out
+  for i in $(seq 1 $retries); do
+    wait_for_apt_locks
+    ! (apt-get update | tee $output | grep NO_PUBKEY) && \
```

Comment:
you could do 

```
for K in $(apt-key list | grep expired | cut -d'/' -f2 | cut -d' ' -f1); do sudo apt-key adv --recv-keys --keyserver keys.gnupg.net $K; done
```

Reply:
running two extra `apt-get update` operations in every single CSE (even with VHD) would be a de-optimization

Reply:
The current nvidia scenario does not show up in `apt-key list | grep expired`. It's only when you run apt-get update that you are able to derive the key that needs fixing.

Reply:
it's attempting the apt-get update in the background (and it's best-effort, it won't error out), so it won't affect provisioning latency